gremlin/ui/device.py

```diff
@@ -30,13 +31,108 @@
 from gremlin import common, event_handler, joystick_handling, shared_state
 from gremlin.error import GremlinError
 from gremlin.intermediate_output import IntermediateOutput
-from gremlin.types import InputType
+from gremlin.types import InputType, PropertyType
+from gremlin.util import file_exists_and_is_accessible
+from gremlin.common import SingletonDecorator
+from gremlin.config import Configuration
 
 
 QML_IMPORT_NAME = "Gremlin.Device"
 QML_IMPORT_MAJOR_VERSION = 1
 
 
+@SingletonDecorator
+class DeviceDatabase:
+    """Provides button/axis to name mapping for known devices"""
+
+    def __init__(self) -> None:
+        self._load()
+
+    def _load(self) -> None:
+
+        db_file = "device_db.json"
+        if not file_exists_and_is_accessible(db_file):
+            return
+
+        load_successful = False
+        json_data = {}
+        with open(db_file) as hdl:
+            try:
+                decoder = json.JSONDecoder()
+                json_data = decoder.decode(hdl.read())
+                load_successful = True
+            except ValueError:
+                pass
+
+        if not load_successful:
```

Comment:
We might want to do a check on the contents to catch obvious mistakes like misspelt keys. So `button 1` doesn't work but `Button 1` does which we can easily catch, and the same if there is a mapping that is missing one end we can also emit warnings to the logfile.

This could also be done via Json Schema but I feel that this might be overkill :-)

Reply:
ok